node/src/actors/rad_manager/mod.rs

```diff
@@ -8,12 +8,45 @@
 
 use crate::utils::stop_system_if_panicking;
 
+use core::iter;
+use itertools::Itertools;
+
 mod actor;
 mod handlers;
 
 /// RadManager actor
 #[derive(Default)]
-pub struct RadManager;
+pub struct RadManager {
+    /// Addresses of proxies to be used as extra transports when performing data retrieval.
+    proxies: Vec<String>,
```

Comment:
Proxy addresses are only validated when executing a data request, so if the user inputs an invalid proxy they have no way to detect it until their node can resolve a data request.

So it would be nice to have either a way to validate the proxies on startup, or a JSON-RPC command to test the execution of a data request using proxies. If the old "try data request" command uses these proxies we can refer to it in the documentation as the expected way to ensure that the node is configured correctly.

Reply:
> So it would be nice to have either a way to validate the proxies on startup

Let me give this a further thought on how to approach it properly.

> or a JSON-RPC command to test the execution of a data request using proxies. If the old "try data request" command uses these proxies we can refer to it in the documentation as the expected way to ensure that the node is configured correctly.

I think I prefer a more automated way :thinking: 

In any case, `try_data_request` is not currently using the proxies (as it implements its own retrieval logic that is not unified with that of `RadManager`). I guess it would be nice to have it use the proxies though, for it to be consistent with the actual witnessing behavior of the node and, as you pointed out, as a funny way to test the proxies functionality.

Reply:
@tmpolaczyk which of these do you think is the right approach to performing validation of proxies upon startup?

1. Validating the addresses
2. Checking that we can connect to a socket
3. Somehow handshaking with the proxy (different for each supported proxy protocol)
4. Running `try_data_request`

Specially in this last case, I think I would give the option of disabling the validation through configuration, so a downtime of the data sources doesn't prevent Witnet nodes from (re)starting.

Reply:
Validating the addresses should be done to detect typos. Checking that we can connect or handshaking may not always work, so I prefer a more end to end test.

When running `try_data_request` from `RadManager` it is really hard to guess which proxy failed, the logs never mention the proxy url, so we need to run `try_data_request` once per proxy, and add a log that clearly says if a proxy failed (and stop the node in case of failure?). Creating a test data request to google.com should work. I guess we can simply add a config variable like `witnessing.http_test_url = "https://google.com"`, and allow `false` to disable that check.

Reply:
Basic URL validation functionality is now in place:

```
$ RUST_LOG=witnet_rad=debug cargo run --release -- -c ~/.witnet/config/witnet.toml wallet server
Loading config from: witnet.toml
Setting log level to: DEBUG, source: Env
[2022-08-11T17:49:25Z INFO  witnet::cli] Sentry telemetry enabled
[2022-08-11T17:49:25Z DEBUG witnet_data_structures] Set environment to mainnet
[2022-08-11T17:49:25Z WARN  witnet_wallet] The local Witnet node JSON-RPC server is configured to listen at 0.0.0.0:21338 but the wallet will connect to 127.0.0.1:21338
[2022-08-11T17:49:26Z INFO  witnet_config::config] The default unproxied HTTP transport for retrieval is enabled.
[2022-08-11T17:49:26Z INFO  witnet_config::config] Configuring retrieval proxies: ["socks5://127.0.0.1", "whatever://192.111.135.17:18302", "laleche"]
[2022-08-11T17:49:26Z INFO  witnet_config::config] Paranoid retrieval percentage is set to 51%
Error: The following transport addresses are invalid:
- socks5://127.0.0.1 (the address is missing a port number)
- whatever://192.111.135.17:18302 ("whatever" is not a supported type of transport)
- laleche (relative URL without a base)
```